From: Hannes Reinecke <hare@suse.de>
Date: Fri, 29 Apr 2016 09:39:01 +0200
Subject: block: don't check request size in blk_cloned_rq_check_limits()
References: bsc#972124
Patch-Mainline: Submitted linux-scsi 2016/05/30

When checking a cloned request there is no need to check
the overall request size; this won't have changed even
when resubmitting to another queue.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 block/blk-core.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/block/blk-core.c b/block/blk-core.c
index 827f8ba..167b1b5 100644
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -2158,11 +2158,6 @@ EXPORT_SYMBOL(submit_bio);
 static int blk_cloned_rq_check_limits(struct request_queue *q,
 				      struct request *rq)
 {
-	if (blk_rq_sectors(rq) > blk_queue_get_max_sectors(q, rq->cmd_flags)) {
-		printk(KERN_ERR "%s: over max size limit.\n", __func__);
-		return -EIO;
-	}
-
 	/*
 	 * queue's settings related to segment counting like q->bounce_pfn
 	 * may differ from that of other stacking queues.
-- 
1.8.5.6

